File: Smack/Transitions/DisplayModalTransitionAnimator.m
//
//  DisplayModalTransitionAnimator.m
//  cabcenter
//
//  Created by Jose Molina Colmenero on 12/06/2017.
//  Copyright © 2017 Frank Lindner. All rights reserved.
//

#import "DisplayModalTransitionAnimator.h"


@implementation DisplayModalTransitionAnimator

- (NSTimeInterval) transitionDuration:(id<UIViewControllerContextTransitioning>)transitionContext {
    return DEFAULT_ANIMATION_DURATION;
}

- (void) animateTransition:(id<UIViewControllerContextTransitioning>)transitionContext {
    UIViewController *fromViewController = [transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
    UIViewController *toViewController = [transitionContext viewControllerForKey:UITransitionContextToViewControllerKey];
    
    toViewController.view.frame = fromViewController.view.frame;
    UIColor *originalColor = toViewController.view.backgroundColor;
    toViewController.view.backgroundColor = [UIColor clearColor];
    
    [transitionContext.containerView addSubview:toViewController.view];
    UIView *firstView = toViewController.view.subviews[0];
    CGRect originalFrame = firstView.frame;
    
    firstView.frame = CGRectMake(firstView.frame.origin.x, fromViewController.view.frame.size.height, firstView.frame.size.width, firstView.frame.size.height);
    
    [UIView animateWithDuration:DEFAULT_ANIMATION_DURATION
                          delay:0
         usingSpringWithDamping:0.8
          initialSpringVelocity:0.1
                        options:UIViewAnimationOptionCurveEaseOut
                     animations:^{
                         firstView.frame = originalFrame;
                         toViewController.view.backgroundColor = originalColor;
                     } completion:^(BOOL finished) {
                         [transitionContext completeTransition:YES];
                     }];
}

@end

File: Smack/Transitions/DisplayModalTransitionAnimator.swift
//
//  DisplayModalTransitionAnimator.swift
//  Smack
//
//  Created by Jose Molina Colmenero on 28/07/2017.
//  Copyright © 2017 Jonny B. All rights reserved.
//

import Foundation

class DisplayModalTransitionAnimator : NSObject, UIViewControllerAnimatedTransitioning {
    
    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
        return 0.5
    }
    
    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
        let fromViewController = transitionContext.viewController(forKey: .from)!
        let toViewController = transitionContext.viewController(forKey: .to)!
        let firstView = toViewController.view.subviews[0]
        let secondView = toViewController.view.subviews[1]
        let originalColor = firstView.backgroundColor
        let originalFrame = secondView.frame

        
        toViewController.view.frame = fromViewController.view.frame
        firstView.backgroundColor = UIColor.clear
        transitionContext.containerView.addSubview(toViewController.view)
        secondView.frame = CGRect(x: firstView.frame.origin.x,
                                 y: fromViewController.view.frame.size.height,
                                 width: firstView.frame.size.width,
                                 height: firstView.frame.size.height)
        
        UIView.animate(withDuration: 0.5,
                       delay: 0,
                       options: .curveEaseOut,
                       animations: {
                        firstView.backgroundColor = originalColor
                        secondView.frame = originalFrame
        }) { (finished) in
            transitionContext.completeTransition(true)
        }
    }
    
}
